_posixsubprocess.c

```diff
@@ -154,6 +154,29 @@ _is_fd_in_sorted_fd_sequence(int fd, PyObject *fd_sequence)
     return 0;
 }
 
+static int
+make_inheritable(PyObject *py_fds_to_keep, int errpipe_write)
+{
+    Py_ssize_t i, len;
+
+    len = PySequence_Length(py_fds_to_keep);
+    for (i = 0; i < len; ++i) {
+        PyObject* fdobj = PySequence_Fast_GET_ITEM(py_fds_to_keep, i);
+        long fd = PyLong_AsLong(fdobj);
+        assert(!PyErr_Occurred());
+        assert(0 <= fd && fd <= INT_MAX);
+        if (fd == errpipe_write) {
+            /* errpipe_write is part of py_fds_to_keep. It must be closed at
+               exec(), but kept open in the child process until exec() is
+               called. */
+            continue;
+        }
+        if (set_inheritable((int)fd, 1, 1, NULL) < 0)
```

Comment:
The third parameter to this call is "raise", and we cannot raise an exception in the only context we call `make_inheritable()` from.  It needs to be 0... but...

I suggest simplifying this patch greatly: Cut out the unnecessary bits of `set_inheritable()` given that the only use of the function is here and should pass `raise=0` and `atomic_flag_works=NULL` which will cut out a lot of unreachable code and get rid of the need for `get_inheritable()` in _posixsubprocess_helpers.c.